eclipse-theia/theia
# mini-browser: improve host pattern logic

#### What it does

Edit our debug config to remove useless env var definitions. Environment variables are already shared between parent/child processes, and there was no real value in the previous definition.

Make the `mini-browser` host pattern logic more resilient in face of empty strings in environment variables. Ultimately it is an edge case but nice to support. Note that on Windows it is not even easily possible to set empty env vars...

See https://github.com/eclipse-theia/theia/pull/9201#issuecomment-800290197

Remove what looked like left-overs in mini-browser. Basically removed `packages/mini-browser/src/electron-main/`. This is marked in the CHANGELOG as a breaking change, although I doubt anyone was relying on it.

#### How to test

Set `THEIA_MINI_BROWSER_HOST_PATTERN=""` before running Theia electron and browser, the mini-browser should work.

On Windows run:

```cmd
node -e "process.env.THEIA_MINI_BROWSER_HOST_PATTERN = ''; child_process.exec('yarn start:electron', { stdio: 'inherit' })"
```

You can open the dev tools and inspect the url for the previewed iframe to use a pattern like `{{uuid}}.mini-browser.{{hostname}}`.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
